### PR TITLE
Set webpack's compilation.warnings when glob-related options are used

### LINKED ISSUE #1380: Use webpack's performance.hints for reporting non-fatal errors
**Library Affected**:
*workbox-webpack-plugin*

**Issue or Feature Request Description**:
`workbox-webpack-plugin` v3 introduced a model in which all assets in the webpack compilation are potentially eligible for precaching (with the `include/exclude/chunks/excludeChunks` options available for modifying the defaults).

Because of this new model, it's (usually) not necessary to use the previous `glob`-based configuration options, which matched files in the output filesystem, rather than matching against assets in the webpack compilation. But there are still some scenarios in which they might be necessary, so we didn't remove support, and documented them at https://developers.google.com/web/tools/workbox/modules/workbox-webpack-plugin#cache_additional_non-webpack_assets

If you do use those `glob`-based options in v3, you end up generating an additional list of precached URLs, and options like `manifestTransforms` and `dontCacheBustUrlsMatching` only apply to that additional list, not the list derived from the webpack asset pipeline.

We should explicitly add one of two possible messages to `compilation.warnings` to let folks know about this scenarios. They are not *always* due to mistakes, so there might be some false positives, but if that's the case, it's easy enough to ignore the warning.

- When `globPatterns` or any of the related options is used, warn that it's often not needed, and link to the relevant docs.
- When `manifestTransforms` or any of the related options that modify the `glob`-derived manifest is used, **and** `globPatterns` is **not** used, warn that those options will have no effect (because there is no `glob`-derived manifest) and link to the relevant docs.

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @addyosmani @gauntface

Fixes #1380

Happy to iterate on the warning language.

This PR also includes updates to the existing test suite to ensure that warnings were not previously being set when they were not expected.

### CODE REVIEW

File: packages/workbox-webpack-plugin/src/lib/warn-about-config.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    'globDirectory',
+    'globFollow',
+    'globIgnores',
+    'globPatterns',
+    'globStrict',
+  ];
+
+  const usedGlobOptions = globOptions.filter((option) => option in config);
+  if (usedGlobOptions.length > 0) {
+    return `You're using the following Workbox configuration options: ` +
```
Comment (gauntface) [inline packages/workbox-webpack-plugin/src/lib/warn-about-config.js]: Maybe `You're using **one** of the following workbox configuration`

File: packages/workbox-webpack-plugin/src/lib/warn-about-config.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      `[${usedGlobOptions.join(', ')}] In Workbox v3 and later, these ` +
```
Comment (gauntface) [inline packages/workbox-webpack-plugin/src/lib/warn-about-config.js]: Add a fullstip after `]`?

File: packages/workbox-webpack-plugin/src/lib/warn-about-config.js (hunk 1 above)
Comment (jeffposnick) [reply packages/workbox-webpack-plugin/src/lib/warn-about-config.js]: The error message only includes the options that are actually being used, so I think it's accurate. The only nit is whether `options` should be singular when there's one option... so I can change that.

File: packages/workbox-webpack-plugin/src/lib/warn-about-config.js (hunk 2 above)
Comment (jeffposnick) [reply packages/workbox-webpack-plugin/src/lib/warn-about-config.js]: Done.

**Pull request merged**